### PR TITLE
refactor ML task data model; add create ML task index method

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
1. Refactor ML task data model
2. Add create ML task index method
 
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by Zhangxunmt — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/model/MLTaskState.java [hunk 1]
```diff
@@ -33,5 +33,6 @@
     CREATED,
     RUNNING,
     COMPLETED,
-    FAILED
+    FAILED,
+    CANCELLED
```
Comment (Zhangxunmt) [inline plugin/src/main/java/org/opensearch/ml/model/MLTaskState.java:37]: Have we used this CANCELLED status anywhere, or this is reserved for future API usage?

Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/model/MLTaskState.java:37]: This is a reserved state

File: plugin/src/main/java/org/opensearch/ml/model/MLTask.java [hunk 2]
```diff
@@ -23,45 +23,161 @@
...
+import org.opensearch.common.xcontent.XContentBuilder;
+import org.opensearch.commons.authuser.User;
+import org.opensearch.ml.common.dataset.MLInputDataType;
+import org.opensearch.ml.common.parameter.FunctionName;
 
 @Getter
 @EqualsAndHashCode
-public class MLTask implements Writeable {
-    private final String taskId;
+public class MLTask implements ToXContentObject, Writeable {
```
Comment (Zhangxunmt) [inline plugin/src/main/java/org/opensearch/ml/model/MLTask.java:34]: We can move this class to the ml-common if we find it necessary to do so in the next PRs.

**Review by Zhangxunmt — APPROVED** [review]
LGTM. If we find necessary we can move the MLTask to ml-common in the next PR.

**Pull request merged**